db/migrate/20191209231307_add_repository_reference_to_stacks.rb

```diff
@@ -0,0 +1,41 @@
+class AddRepositoryReferenceToStacks < ActiveRecord::Migration[6.0]
+  def up
+    add_reference :stacks, :repository
+
+    repositories = Shipit::Stack.distinct.select(:repo_owner, :repo_name).pluck(:repo_owner, :repo_name)
+    repositories.map do |repo_owner, repo_name|
+      repository = Shipit::Repository.create_or_find_by(
+        owner: repo_owner,
+        name: repo_name,
+      )
+
+      stacks = Shipit::Stack.where(repo_owner: repository.owner, repo_name: repository.name)
+      stacks.update(repository: repository)
+    end
```

Comment:
I typically like to restrict Migrations to DDL and perform DML in rake tasks as part of a multi-phase "Roll Out", but I'm not sure what is prudent here. Any guidance would be greatly appreciated.

Reply:
Historically we've been doing DML in migrations. I agree it's not ideal, but it makes it much easier for the users.